lib/active_merchant/billing/gateways/plexo.rb

```diff
@@ -0,0 +1,260 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class PlexoGateway < Gateway
+      self.test_url = 'https://api.testing.plexo.com.uy/v1/payments'
+      self.live_url = 'https://api.plexo.com.uy/v1/payments'
+
+      self.supported_countries = ['UY']
+      self.default_currency = 'UYU'
+      self.supported_cardtypes = %i[visa master american_express discover]
+
+      self.homepage_url = 'https://www.plexo.com.uy'
+      self.display_name = 'Plexo'
+
+      APPENDED_URLS = %w(captures refunds cancellations)
+
+      def initialize(options = {})
+        requires!(options, :client_id, :api_key)
+        @credentials = options
+        super
+      end
+
+      def purchase(money, payment, options = {})
+        response = MultiResponse.run do |r|
+          r.process { authorize(money, payment, options) }
+          r.process { capture(money, r.authorization, options) }
```

Comment:
What happens here if the `authorize` fails?

Reply:
It should be able to respond with error as expected. Here is the test to cover that case:

https://github.com/activemerchant/active_merchant/pull/4432/files#diff-b724dc322c2cc0818829951d8a93bb9fddbffd55fd3a717be266ecffe9d000a6R44

Reply:
Great, thanks! I went looking for the test and just didn't find it.